x-pack/plugins/ml/public/settings/filter_lists/components/filter_list_items_grid/filter_list_items_pagination.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+/*
+ * React component for the pagination controls of the filter list items grid.
+ */
+
+import PropTypes from 'prop-types';
+import React, {
+  Component,
+} from 'react';
+
+import {
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPagination,
+  EuiPopover,
+} from '@elastic/eui';
+
+
+function getContextMenuItemIcon(menuItemSetting, itemsPerPage) {
+  return (menuItemSetting === itemsPerPage) ? 'check' : 'empty';
+}
+
+
+export class FilterListItemsPagination extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isPopoverOpen: false
+    };
+  }
+
+  onButtonClick = () => {
+    this.setState({
+      isPopoverOpen: !this.state.isPopoverOpen,
+    });
+  }
+
+  closePopover = () => {
+    this.setState({
+      isPopoverOpen: false,
+    });
+  }
+
+  onPageClick = (pageNumber) => {
+    this.props.setActivePage(pageNumber);
+  }
+
+  onChangeItemsPerPage = (pageSize) => {
+    this.closePopover();
+    this.props.setItemsPerPage(pageSize);
+  }
+
+  render() {
+
+    const {
+      itemCount,
+      itemsPerPage,
+      activePage } = this.props;
+
+    const button = (
+      <EuiButtonEmpty
+        size="s"
+        color="text"
+        iconType="arrowDown"
+        iconSide="right"
+        onClick={this.onButtonClick}
+      >
+        Items per page: {itemsPerPage}
+      </EuiButtonEmpty>
+    );
+
+    const pageCount = Math.ceil(itemCount / itemsPerPage);
+
+    const items = [
```

Comment:
Just a suggestion to reduce redundancy: This could be refactored to use a stateless functional component or map over an Array like `[50, 100, 500, 100].map(items => (<Eui ... >)`